nilearn/regions/region_extractor.py

```diff
@@ -0,0 +1,320 @@
+"""
+Better brain parcellations for Region of Interest analysis
+"""
+import warnings
+import numbers
+import nibabel
+import numpy as np
+
+from scipy.ndimage import label
+
+from sklearn.base import clone
+from sklearn.externals.joblib import Memory
+
+from .. import masking
+from ..input_data import NiftiMapsMasker
+from .._utils import check_niimg, check_niimg_4d
+from ..image import new_img_like, resample_img
+from ..image.image import _smooth_array, threshold_img
+from .._utils.niimg_conversions import concat_niimgs, _check_same_fov
+from .._utils.compat import _basestring
+from .._utils.ndimage import _peak_local_max
+from .._utils.segmentation import _random_walker
+
+
+def _threshold_maps(maps, threshold):
+    """ Automatic threshold function using ratio to the number of voxels
+    in the brain volume to keep the most prominent features of the maps
+    which lies above a certain threshold
+    """
+    maps = check_niimg(maps)
+    maps_data = maps.get_data()
+    abs_maps = np.abs(maps_data)
+    if isinstance(threshold, float):
+        ratio = threshold
+    elif threshold is not None:
+        raise ValueError("Threshold must be float value for "
+                         "thresholding_strategy='ratio_n_voxels'. "
+                         "You provided %s." % str(threshold))
+
+    raveled = abs_maps.ravel()
+    argsort = np.argsort(raveled)
+    n_voxels = (ratio * maps_data.size)
+    cutoff_threshold = raveled[argsort[- n_voxels]]
+    maps_data[abs_maps < cutoff_threshold] = 0.
+
+    threshold_maps = new_img_like(maps, maps_data)
+
+    return threshold_maps
+
+
+def connected_regions(maps_img, min_size=50,
+                      extract_type='local_regions',
+                      peak_local_smooth=6, mask_img=None):
+    """ A function takes the connected components of the brain activation
```

Comment:
"Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description" from [PEP 0257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings).

That would be nice if you could fix this in other docstrings as well.